### DISCUSSION TITLE
Teamwork with Nextcloud and QField Cloud

### DISCUSSION THREAD

**Discussion by thymaro** [opengisch/QField#3271] [Q&A]
**This might be a bit convoluted, I'm sorry, as there's a lot to unpack. If something is unclear, just ask away, and I will do my best to clarify in an edit or an answer.**

While working with NC (Nextcloud) and QFC (QField Cloud), our team encountered some unresolved questions.

We use NC to store our data, projects, just about anything we do and create on computers. We use printed field maps and printed field tables (lists, i.e. attribute tables, essentially!) to record our data, then go back to the office, scan the field maps and digitize them. The process is a bit more complicated, but this should do for an MWE. Some time ago, we tried moving that process to QField by cable sync, but that didn't really yield enough benefits. Recently, I started making QFC projects for testing and asked my boss to download the project to their device. Everything smooth, so far.

Process and Setup:

- QGIS project with TIFF and GPKG. Hosted on our NC instance and synced to our individual devices.
- QGIS project exported to a QFC project in a subdirectory in the project dir on NC, while uploading the project to QFC servers.
- Collect data outside with a tablet and push the changes to QFC Cloud. Also, note what could be improved in terms of layer styling and themes. (**this is very important to understand the issue further down**)
- Back at the office, open QGIS and sync from the QFC server with the plugin. After the sync, there is a dialog asking whether or not to open the project. Open the project and the “Project name (QField Cloud)” opens.
- Open original project and **wish to** sync the freshly recorded field data. Stop and wonder whether that even works.
- Stop and wonder where to make styling changes: to the original QGIS project “Project name” or to the “Project name (QField Cloud)”?

One of the problems is finding out who should do the syncing to the “local” (i.e. on NC) projects. **Can anyone do this, or does it have to be the “owner” of the project?**

**Can we import the QFC changes into the original project?** The problem there is that we have a specific file structure for GIS projects and, of course, the file structure in the exported QFC project is completely different, everything being in one and the same folder.

**If we open the original project and make styling, layout, theming changes to it, what is the best process to sync these changes to the QFC server?** I imagine that if I delete the QFC project on disk, make changes to the QGIS project, export to QFC: it will erase the data that will have been collected in the meantime.

**In general: What is the recommended workflow to work with NC and QFC projects?**

_aside: I have suggested moving some layers to PostGIS to have access to them from anywhere, but they are not keen on using PostGIS, yet, so I'll have to work on that more. In the meantime, I bought some server space for another private project and discovered they offer postgres and postgis, so I'll convert some of my layers to PostGIS tables. This way, I don't have to worry about changes in some GPKG layers not being synced. For right now, this will suffice. I can just wipe the QFC project, make changes to the original QGIS project and push that to a new QFC project without fear of dataloss. It would still be nice to know how to do a similar thing with QFC, though._

**Comment by Pellethuer** [comment]
I have an different setup, not using QFC and sync directly with the NC:

- Accessing the NC server via WEB-DAV on the individual localdevice in the office
- one NC folder for one mobile device containing all the needed QF projects for this device
- packing the QF project to this NC folder
- sync NC folder directly with the QF folder on my mobile device, using the app foldersync. This works, if you synchronice the correct folder on your mobile device. My mobile stores the project to the sd card in the folder: storage/6639-6263/Android/data/ch.opengis.qfield/files/importet Projects *

Workflow
- QGIS project 
- packing QF project on NC folder connected with WEB DAV
- sync QF folder and mobile device folder using the app "foldersync" on the mobile device*
- Collecting data in the field
- sync mobile device folder with the NC folder using "foldersync" again.
- import QF project from the NC folder in the office

*first time I copied the QF project on my mobile in the download folder and opened it in QF using "open local files".  Sync didn't work. So I fingered out, that QF stored the Project on the sd card in the file storage/6639-.... Synchronizing that folder works.

**Reply by thymaro** [reply]
Am I understanding this right, that it won't work as soon as multiple people are working on several mobile devices simultaneously. Even if they pay great care to not sync to the same folder in NC, when you open the QGIS project and import the QField data from person A, then importing the QField data from person B will corrupt things, right?
